braindecode/models/sinc_shallow.py

```diff
@@ -0,0 +1,338 @@
+    """Sinc-Based Convolutional Layer for Band-Pass Filtering from Ravanelli and Bengio (2018) [ravanelli]_.
+
+    The `SincFilter` layer implements a convolutional layer where each kernel is
+    defined using a parametrized sinc function.
+    This design enforces each kernel to represent a band-pass filter,
+    reducing the number of trainable parameters.
+
+    Parameters
+    ----------
+    low_freqs : torch.Tensor
+        Initial low cutoff frequencies for each filter.
+    kernel_size : int
+        Size of the convolutional kernels (filters). Must be odd.
+    sfreq : float
+        Sampling rate of the input signal.
+    bandwidth : float, optional
+        Initial bandwidth for each filter. Default is 4.0.
+    min_freq : float, optional
+        Minimum frequency allowed for low frequencies. Default is 1.0.
+    padding : str, optional
+        Padding mode, either 'same' or 'valid'. Default is 'same'.
+
+    References
+    ----------
+    .. [ravanelli] Ravanelli, M., & Bengio, Y. (2018, December). Speaker
+       recognition from raw waveform with sincnet. In 2018 IEEE spoken language
+       technology workshop (SLT) (pp. 1021-1028). IEEE.
+    """
+
+    def __init__(
+        self,
+        low_freqs: torch.Tensor,
+        kernel_size: int,
+        sfreq: float,
+        bandwidth: float = 4.0,
+        min_freq: float = 1.0,
+        padding: str = "same",
+    ):
+        super().__init__()
+        if kernel_size % 2 == 0:
+            raise ValueError("Kernel size must be odd.")
+
+        self.num_filters = low_freqs.numel()
+        self.kernel_size = kernel_size
+        self.sfreq = sfreq
+        self.min_freq = min_freq
+        self.padding = padding.lower()
+
+        # Precompute constants
+        window = torch.hamming_window(kernel_size, periodic=False)
+
+        self.register_buffer("window", window[: kernel_size // 2].unsqueeze(-1))
+
+        n_pi = (
+            torch.arange(-(kernel_size // 2), 0, dtype=torch.float32)
+            / sfreq
+            * 2
+            * math.pi
+        )
+        self.register_buffer("n_pi", n_pi.unsqueeze(-1))
+
+        # Initialize learnable parameters
+        bandwidths = torch.full((1, self.num_filters), bandwidth)
+        self.bandwidths = nn.Parameter(bandwidths)
+        self.low_freqs = nn.Parameter(low_freqs.unsqueeze(0))
+
+        # Constant tensor of ones for filter construction
+        self.register_buffer("ones", torch.ones(1, 1, 1, self.num_filters))
+
+    def build_sinc_filters(self) -> torch.Tensor:
+        """Builds the sinc filters based on current parameters."""
+        # Computing the low frequencies of the filters
+        low_freqs = self.min_freq + torch.abs(self.low_freqs)
+        # Setting a minimum band and minimum freq
+        high_freqs = torch.clamp(
+            low_freqs + torch.abs(self.bandwidths),
+            min=self.min_freq,
+            max=self.sfreq / 2.0,
+        )
+        bandwidths = high_freqs - low_freqs
```

Comment:
normalization matters here. 

Get the model after the training and before (check if the random stage initialization is the same).

Reply:
Already solve on EEGMiner.
https://github.com/braindecode/braindecode/blob/cf53ebc1e1805913c5331dfc0711d78f2e300f12/braindecode/models/eegminer.py#L388